app/Http/Controllers/Api/VenueController.php

```diff
@@ -116,9 +123,55 @@ public function edit(Venue $venue)
      */
     public function update(Request $request, Venue $venue)
     {
-        //
     }
 
+    public function rate_venue(Request $request)
+    {
+        try {
+            //Get venue, rating and user
+            $venue = Venue::findOrFail($request->venue_id);
+            $user = User::findOrFail($request->user_id);
+
+            $user_rating = DB::table('ratings')
+                                ->where('user_id', '=', $user->id)
+                                ->where('rateable_id', '=', $venue->id)
+                                ->first();
+
+            if ($user_rating) {
```

Comment:
you might be able to use firstOrCreate here - https://laravel-news.com/firstornew-firstorcreate-firstor-updateorcreate